### PR TITLE
Cloak grenades only stealth marines, supplies cut in half

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request
Cloak grenades now only stealth marines. Cut cloak grenade supply in half to account for this.

## Why It's Good For The Game
Makes cloak grenades easier to play around and can help as an option that encourages pushing by allowing marines to advance while being harder to be focused down or ganged up on by xenos due to being stealthed. Xenos can still see the smoke, so marines using these aren't completely unpredictable either. 

Having xenos be stealthed like currently is much better for xenos than marines being stealthed since xenos can generally tank a few stray shots and most of their skills are projectiles, have a good area of effect or bumpslash, meaning they don't need to specifically see their target to be effective or worry about FF.

## Changelog
:cl:
balance: cloak grenades cloak only marines instead of marines and xenos, decreased alpha by ~70%
balance: cut cloak grenade supplies in half
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -232,10 +232,10 @@
 /mob/living/effect_smoke(obj/effect/particle_effect/smoke/S)
 	. = ..()
 	if(!.)
-		if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO))
+		if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO) && ishuman(src))
```
Comment (BraveMole) [inline code/modules/mob/living/living_defense.dm:235]: istype(src) == bad. Use flags or even better, override the proc

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -232,10 +232,10 @@
...
 			smokecloak_off()
 		return
-	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO))
+	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO) && ishuman(src))
```
Comment (TiviPlus) [inline code/modules/mob/living/living_defense.dm:238]: Balance wise turned out to be fine, fix this

File: code/modules/mob/living/living_defense.dm (hunk 1 above)
Comment (Walarks) [reply code/modules/mob/living/living_defense.dm]: moved to human/effect_smoke

File: code/modules/mob/living/living_defense.dm (hunk 2 above)
Comment (Walarks) [reply code/modules/mob/living/living_defense.dm]: moved to human/effect_smoke

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**